config.js

```diff
@@ -151,23 +116,35 @@ exports.updateFromArgv = function(argv) {
     // console messages
     // C - test exports warnings
 
-    testExportsWarnings();
+    warnIfNopauseIsMissing("ranked");
+    warnIfNopauseIsMissing("unranked");
+    console.log("\n");
 
 }
 
+function exportInitialConfigRankedUnranked(rankedUnrankedUnderscored) {
+    exports[`banned_users${rankedUnrankedUnderscored}`] = {};
+    exports[`allow_all_boardsizes${rankedUnrankedUnderscored}`] = false;
+    exports[`allowed_boardsizes${rankedUnrankedUnderscored}`] = [];
+    exports[`allow_all_komis${rankedUnrankedUnderscored}`] = false;
+    exports[`allowed_komis${rankedUnrankedUnderscored}`] = [];
+    exports[`allowed_speeds${rankedUnrankedUnderscored}`] = {};
+    exports[`allow_all_speeds${rankedUnrankedUnderscored}`] = false;
+    exports[`allowed_timecontrols${rankedUnrankedUnderscored}`] = {};
+    exports[`allow_all_timecontrols${rankedUnrankedUnderscored}`] = false;
+}
+
 function testRankedUnrankedOptions(rankedUnrankedOptions, argv) {
     for (const option of rankedUnrankedOptions) {
         const [general, ranked, unranked] = getArgNamesGRU(option.name);
         
         // check undefined specifically to handle valid values such as 0 or null which are tested false
         if (argv[general] !== undefined) {
             if (argv[ranked] !== undefined) {
-                throw `Cannot use --${general} and --${ranked} at the same time.\nFor ranked games, `
-                      + `use either --${general} or --${ranked} or no option if you want to allow all values.`;
+                throw `Cannot use --${general} and --${ranked} at the same time.\nFor ranked games, use either --${general} or --${ranked} or no option if you want to allow all values.`;
```

Comment:
No option means we use default right?

Reply:
the `or no option` in `or no option if you want to allow all values` was not clear indeed, plus it does not always allow all values, it allows all values only if the option's default is empty

i replaced it with `or do not use anything to switch to default settings for this option`
it is indeed a lot clearer :)